elooplib/client.cpp

```diff
@@ -1,131 +1,187 @@
+    : Client(cl.getFd()), th_pool(tp), event_fd(fd)
 {
-    response = response.substr(9, request_field.size() - 10);
-    std::cout << response << std::endl; // DEBUG
-    const std::filesystem::path dir(response);
-    if (!exists(dir))
-    {
-        response = "Path is not exists\n";
-        return;
-    }
+  dir = cl.getResponse();
+  ClientTask::compressFiles();
+}
 
-    int dir_len = 0;
-    // TODO try_lock if(1) etc
-    std::list<std::string> files;
-    for (const auto &entry : std::filesystem::directory_iterator(response))
-    {
-        std::string tmp = entry.path();
-        if(tmp.find(".gz") == std::string::npos)
-        {
-            //std::remove_if(begin(tmp), end(tmp), '\"'); TODO
-            files.push_back(tmp); 
+ClientTask::ClientTask(ClientTask &&rhs)
+    : Client(static_cast<Client &&>(rhs)), th_pool(std::move(rhs.th_pool)) {}
 
-        }
+int ClientTask::compressFiles()
+{
 
-        dir_len++;
+  // TODO try_lock for directory name mutex
+  int task_summ = 0;
+  for (const auto &entry : std::filesystem::directory_iterator(dir))
+    task_summ++;
+  for (const auto &entry : std::filesystem::directory_iterator(dir))
+  {
+    std::string tmp = entry.path();
+    if (!tmp.ends_with(".gz"))
+    {
+      auto ct = std::shared_ptr<ITask>(
+          new CompressITask(tmp, ClientTask::getFd(), &complete_tasks,
+                            &success_tasks, task_summ));
+      th_pool->addTask(ct);
     }
-    
-    files_counter = 0;
-    for(const auto &f: files)
-        compressFile(f);
-
-    response = std::to_string(files_counter);
+  }
 }
 
-void Client::compressFile(const std::string &in_file) // TODO
+int ClientTask::handleConnection()
 {
-    /*int res = compress(in_file.c_str(), (in_file + ".gz").c_str(),
-                       6);
-    if (res == Z_OK)
-        files_counter++; */
+  dir = std::to_string(success_tasks) + "\n";
+  int s = send(Node::getFd(), dir.c_str(), dir.size(), 0);
+  if (s < 0)
+    throw serverExcept("send()");
+  int parent_fd = Node::getFd();
+  ClientTask::closeConnection();
+  return parent_fd;
 }
-void Client::sendData()
+
+void ClientTask::closeConnection()
 {
-    recognizeData();
-    int s = send(getFd(), response.c_str(), response.size(), 0);
-    if (s < 0)
-        throw serverExcept("send()");
+  Node::closeConnection();
+  setFd(-1); // needed to keep socket opened after destructor call
 }
 
+size_t ClientTask::getFd() { return event_fd; }
+ClientTask::~ClientTask() { close(event_fd); }
+/////////
+CompressITask::CompressITask(const std::string &f, int fd, std::atomic<int> *tn,
+                             std::atomic<int> *s, int m)
+    : ITask(), filename(f), event_fd(fd), task_num(tn), success(s), max(m){};
+
+void CompressITask::run() { compressFile(); }
 
+int CompressITask::compressFile()
+{
+  // TODO compression
+  using namespace std::chrono_literals;
+  std::this_thread::sleep_for(6000ms); // DEBUG
```

Comment:
Tip: you can use defines for this purpose:
`#ifdef BVN_DEBUG`. And set it in Makefile optionally